### PR TITLE
Add InfoBar Notification Requested

### PR DESCRIPTION
**Pull request by rafael-rosa-knowcode**
## PR Type

What kind of change does this PR introduce?

- Feature

## What is the current behavior?

We do not have the Notification on the Visual Studio/DevServer

## What is the new behavior?

To be able to add notification bar on the VS.

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [ ] Contains **NO** breaking changes
- [ ] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [ ] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

## Other information

Internal Issue (If applicable):

### CODE REVIEW

File: src/Uno.UI.RemoteControl.VS/EntryPoint.cs [hunk 1]
```diff
@@ -419,6 +422,68 @@ async Task Restart()
...
+		if (infoBar is not null)
+		{
+			infoBar.ActionItemClicked += (s, e) =>
+			{
+				_asyncPackage.JoinableTaskFactory.Run(async () =>
+				{
+					if (e.ActionItem is ActionBarItem action &&
+						action.Name is { } command)
+					{
+						var cmd =
```
Comment (rafael-rosa-knowcode) [inline src/Uno.UI.RemoteControl.VS/EntryPoint.cs:474]: Using process ID.

File: src/Uno.UI.RemoteControl.VS/EntryPoint.cs [hunk 2]
```diff
@@ -419,6 +422,68 @@ async Task Restart()
...
+							new CommandRequestIdeMessage(
+								System.Diagnostics.Process.GetCurrentProcess().Id,
+								command,
+								action.ActionContext?.ToString());
+						await _ideChannelClient.SendToDevServerAsync(cmd, _ct.Token);
+					}
+				});
```
Comment (rafael-rosa-knowcode) [inline src/Uno.UI.RemoteControl.VS/EntryPoint.cs:481]: Using Process ID

File: src/Uno.UI.RemoteControl.VS/EntryPoint.cs [hunk 3]
```diff
@@ -10,18 +10,19 @@
 using System.Reflection;
 using System.Threading;
 using System.Threading.Tasks;
-using System.Windows.Forms;
 using EnvDTE;
 using EnvDTE80;
 using Microsoft.Build.Evaluation;
 using Microsoft.Internal.VisualStudio.Shell;
+using Microsoft.VisualStudio.Imaging;
```
Comment (dr1rrb) [inline src/Uno.UI.RemoteControl.VS/EntryPoint.cs:17]: Is this namespace really needed ? Sounds weird

File: src/Uno.UI.RemoteControl.VS/EntryPoint.cs [hunk 4]
```diff
@@ -344,6 +345,7 @@ private async Task EnsureServerAsync()
 
 				_ideChannelClient = new IdeChannelClient(pipeGuid, new Logger(this));
 				_ideChannelClient.ForceHotReloadRequested += OnForceHotReloadRequestedAsync;
+				_ideChannelClient.InfoBarNotificationRequested += OnInfoBarNotificationRequestedAsync;
```
Comment (dr1rrb) [inline src/Uno.UI.RemoteControl.VS/EntryPoint.cs:348]: I would have create a generic event `OnMessageReceived`, but I'm fine to keep it this way for now.

File: src/Uno.UI.RemoteControl.VS/IDEChannel/IDEChannelClient.cs [hunk 5]
```diff
@@ -3,11 +3,16 @@
 using System.Linq;
 using System.Threading;
 using System.Threading.Tasks;
+using EnvDTE;
+using Microsoft.VisualStudio.Imaging;
+using Microsoft.VisualStudio.Shell;
+using Microsoft.VisualStudio.Shell.Interop;
```
Comment (dr1rrb) [inline src/Uno.UI.RemoteControl.VS/IDEChannel/IDEChannelClient.cs:9]: Same here, I think we don't need all those namespaces?

File: src/Uno.UI.RemoteControl.VS/EntryPoint.cs (hunk 2 above)
Comment (dr1rrb) [reply src/Uno.UI.RemoteControl.VS/EntryPoint.cs:480]: That's fine for now, please rename the property accordingly then (cc. @jeromelaban )

**Review by dr1rrb — APPROVED** [review]

File: src/Uno.UI.RemoteControl.VS/EntryPoint.cs (hunk 3 above)
Comment (rafael-rosa-knowcode) [reply src/Uno.UI.RemoteControl.VS/EntryPoint.cs:17]: We need the namespace for the [KnownMonikers ](https://learn.microsoft.com/en-us/dotnet/api/microsoft.visualstudio.imaging.knownmonikers?view=visualstudiosdk-2022)

File: src/Uno.UI.RemoteControl.VS/EntryPoint.cs (hunk 4 above)
Comment (rafael-rosa-knowcode) [reply src/Uno.UI.RemoteControl.VS/EntryPoint.cs]: Done. Updated. https://github.com/unoplatform/uno/commit/fa522e3b62af77d4c097808634ac8a9fbe085efa

File: src/Uno.UI.RemoteControl.VS/IDEChannel/IDEChannelClient.cs (hunk 5 above)
Comment (rafael-rosa-knowcode) [reply src/Uno.UI.RemoteControl.VS/IDEChannel/IDEChannelClient.cs]: Done. Removed. https://github.com/unoplatform/uno/commit/da8d2feb4472187808112622109a6ec1819a1004

**Comment by rafael-rosa-knowcode** [comment]
I changed back to the WindowID of the Process the Visual Studio is running. 
https://github.com/unoplatform/uno/commit/c7dc816ce18e62248f826abca97a7a9307696abd

**Comment by jeromelaban** [comment]
@mergifyio backport release/stable/5.5

**Review by ajpinedam — APPROVED** [review]

**Review by dr1rrb — APPROVED** [review]

File: src/Uno.UI.RemoteControl.VS/IDEChannel/IDEChannelClient.cs [hunk 6]
```diff
@@ -98,6 +99,10 @@ private void ProcessDevServerMessage(object sender, IdeMessageEnvelope devServer
 				case KeepAliveIdeMessage:
 					_logger.Verbose($"Keep alive from Dev Server");
 					break;
+				case NotificationRequestIdeMessage e when e is { } message:
+					_logger.Verbose($"Dev Server will open the Notification Message with message {e.Message}");
+					process = OnMessageReceived.InvokeAsync(this, message);
```
Comment (dr1rrb) [inline src/Uno.UI.RemoteControl.VS/IDEChannel/IDEChannelClient.cs:104]: The idea to have a generic `OnMessageReceived` was to remove dependency to all type of message from the `IDEChannelClient` (i.e. remove that switch case). But that's fine for now, let merge this PR and address it later.

Comment (dr1rrb) [reply src/Uno.UI.RemoteControl.VS/IDEChannel/IDEChannelClient.cs:104]: @rafael-rosa-knowcode

**Pull request merged**